uhubctl.c

```diff
@@ -1099,6 +1112,12 @@ int main(int argc, char *argv[])
     int rc;
     int c = 0;
     int option_index = 0;
+#if defined(LIBUSB_API_VERSION) && (LIBUSB_API_VERSION >= 0x01000107) && \
+  (defined(__gnu_linux__) || defined(__linux__))
+    int sys_fd;
+    libusb_device_handle *sys_devh = NULL;
+    libusb_device *dev_list[2];
```

Comment:
Please move this variable definition into if (opt_sysdev) block - the only place it is used

Reply:
The allocated memory is pointed to by the usb_dev pointer later, so it cannot be free'd before libusb_free_device_list(usb_devs, 1) at the end.

Reply:
So it cannot be allocated on the stack inside the if (opt_sysdev) block. I agree this isn't pretty. The alternative is malloc'ing it inside that block instead, but I found it a lot of overhead for 2 measly pointers.

Reply:
Let me know what you think about the calloc variant I pushed now.